### PR TITLE
fix(pageable): hoist non-pseudo position:absolute children out of flow

### PR DESCRIPTION
**Pull request by mitsuru**
## 概要

`` / `<div position:absolute>` 等の **非疑似** 絶対配置要素が pagination 中に in-flow 扱いになり、要素高がページ高を超える場合に独立ページを占有していた問題 (`fulgur-aijf`) を修正。

CSS 2.1 §10.3.7 / §10.6.4 に従い、abs/fixed 子要素を out-of-flow として containing block 相対位置で再 emit する。

### 修正前
```html
<style>@page { size: 300px 50px; margin: 0; }</style>
<img style="position:absolute; top:0; left:0;" src="cat.png">
Cat head.
<div style="break-before:page;">Cat body.</div>
<div style="break-before:page;">No cat parts on this page.</div>
```
→ 4 ページ (page 1 = 画像のみ、page 2-4 = 本文)

### 修正後
→ 3 ページ (画像は out-of-flow、break-before の 3 div が page count を決定)

### 1. `PositionedChild::out_of_flow` フラグ

`BlockPageable::wrap` で out-of-flow children を `total_height` 計算から除外し、`find_split_point` の overflow scan / forced-break probe / 主 scan loop からも除外。`split` / `split_boxed` では新ヘルパ `split_children_at_index` / `split_children_for_within` で両 fragment に複製。`clone_pc_with_offset` は out-of-flow に対して y クランプを skip し、CB 相対座標を保持して Krilla の page-area クリップで自然に slicing。

既存の pseudo path にも `out_of_flow: true` を流し込み、tall abs pseudo で再現する latent bug を予防。

### 2. `build_absolute_non_pseudo_children`

`build_absolute_pseudo_children` をミラーする形で `node` の DOM 直接子を走査し、`is_absolutely_positioned` な非疑似要素を `PositionedChild { out_of_flow: true }` として hoist。CB resolve (`resolve_cb_for_absolute` / `cb_padding_box`) と inset 解決ロジックは pseudo 版と共有。

`build_absolute_children` wrapper 経由で 5 つの call site に配線:

- `convert/inline_root.rs` ×2
- `convert/list_item.rs` ×2
- `convert/pseudo.rs` (`wrap_with_pseudo_content`)

`collect_positioned_children` で abs 子を skip して二重 emit を防止。

### スコープ制限

abs 要素は **直接の親に emit** する (pseudo path と同一 semantics)。深くネストした abs (CB が複数階層上にあるケース) は別 issue で対応。`page-background-002/003` (abs が body 直下) には十分。

## 検証

- **fulgur lib tests**: 828 passed (baseline 824 + 新 helper 4)
- **新 regression test** (`tests/abs_positioned_pagination.rs`): 2 件
- **integration / VRT / fulgur-cli**: 全緑
- **fmt / clippy**: 緑

### WPT (css-page) delta

| | main | this PR | delta |
|:-:|:-:|:-:|:-:|
| pass | 45 | 46 | +1 |
| regressions | 17 | 17 | ±0 |
| promotions | 7 | 0 | -7 (= 既 promote) |

- **+3 newly PASSing**: `page-background-002` / `page-background-003` (受け入れ基準) + `fixedpos-001`
- **-2 newly FAILing** (follow-up `fulgur-puml` で tracking):
  - `monolithic-overflow-013`: 350vh の abs 内容がページ生成を駆動できない (以前は abs-as-in-flow で偶然 4 ページになっていた)
  - `page-name-propagated-003`: 142/891662 px の微差 (abs が in-flow space を確保しないため周辺レイアウトが sub-pixel 単位で移動)
- 加えて main で既に observed=PASS だった 7 件を expectations 上で `PASS` に昇格

## 受け入れ基準

- [x] 再現 HTML の page count が 3 になる
- [x] `page-background-002/003` が PASS
- [x] 既存テスト緑
- [x] CSS 2.1 §10.3.7 / §10.6.4 準拠の helper unit test 追加

Closes fulgur-aijf (merge 後)

## Test plan

- [ ] CI green (lib / integration / VRT / fulgur-cli / WPT)
- [ ] fmt / clippy clean
- [ ] WPT regressions/promotions delta 0 (expectations と一致)

---

<a href="https://app.devin.ai/review/fulgur-rs/fulgur/pull/260" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Resolved multiple CSS print/page formatting issues (fixed positioning, page backgrounds, page names, page orientation and page-size handling)
  * Improved pagination so absolutely-positioned content no longer consumes page space or is silently dropped

* **Tests**
  * Added regression tests for absolute-element pagination and flattening edge cases
  * Updated web-platform test expectations: many print tests marked passing; two tests noted as regressions (including a small pixel-diff failure)

### CODE REVIEW

File: crates/fulgur/src/pageable.rs
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/pageable.rs:1]: 85c97bf で修正しました。`find_split_point` の冒頭で in-flow children のメタデータを事前計算するように変更しています:

- `first_in_flow_index` / `last_in_flow_index`: in-flow children の最初/最後の index
- `in_flow_count`: in-flow children の総数
- `next_in_flow_at_or_after(from)`: 指定 index 以降で最初の in-flow を返すヘルパ
- `i_is_first_in_flow(i)` / `i_is_last_in_flow(i)` / `has_in_flow_before(i)` / `has_in_flow_after(i)` クロージャ

ご指摘の 3 点はそれぞれ次のように対応:

1. **single-overflowing-child guard** (`i == 0 && self.children.len() == 1`) → `in_flow_count == 1` に変更。OOF children の混在に関わらず、in-flow が 1 件のみのときに `NoSplit` を返します。
2. **break_after guard** (`i < self.children.len() - 1`) → `has_in_flow_after(i)` に変更。OOF が末尾に追加されていても、in-flow としての「最後」を正しく判定します。
3. **break_before guard** (`i > 0`) → `!i_is_first_in_flow(i)` に変更。OOF が先頭にあるケースでも誤発火しません。

加えて、`AtIndex` ターゲットが OOF を指す問題には:

- break_after の `AtIndex(i + 1)` → `next_in_flow_at_or_after(i + 1)` で trailing OOF を skip
- overflow fallback の `AtIndex(i.max(1))` → 同様に `next_in_flow_at_or_after(i.max(1))` で skip
- 後続 in-flow が存在しない場合は `NoSplit` を返す

`split_y = self.children[idx].y` が常に in-flow の y 座標を読むようになります。回帰テスト `break_after_on_last_in_flow_with_trailing_abs_sibling_does_not_split` を追加しました (break-after が最後の in-flow に付与され trailing abs sibling がいるケース、修正前は spurious 2 page split していたもの)。

[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/pageable.rs:1]: 85c97bf で修正しました。`BlockPageable::has_forced_break_below` で out_of_flow children を skip するように変更しています (CSS 2.1 §10.6.4 — abs/fixed の pagination は親の flow と独立)。これで OOF descendant に `break_before:page` / `break_after:page` があっても親の `has_forced_break_below()` が誤って `true` を返さなくなり、ご指摘の「ancestor blocks に spurious split を誘発する」経路が閉じます。

File: crates/fulgur/src/convert/positioned.rs [hunk 1]
```diff
@@ -427,15 +459,153 @@ pub(super) fn build_absolute_pseudo_children(
...
+pub(super) fn build_absolute_children(
+    doc: &blitz_dom::BaseDocument,
+    node: &Node,
+    ctx: &mut ConvertContext<'_>,
+    depth: usize,
+) -> Vec<PositionedChild> {
+    let mut out = build_absolute_pseudo_children(doc, node, ctx, depth);
+    out.extend(build_absolute_non_pseudo_children(doc, node, ctx, depth));
+    out
+}
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/convert/positioned.rs:625]: d338672 で修正しました。`build_absolute_pseudo_children` を slot を取れる形に変更 (引数 `slots: &[Option<usize>]`) し、`build_absolute_children` で次の順に出力するように:

1. `build_absolute_pseudo_children(doc, node, ctx, depth, &[node.before])` — `::before` 疑似 (もし abs の場合)
2. `build_absolute_non_pseudo_children(doc, node, ctx, depth)` — 直接の DOM abs/fixed 子
3. `build_absolute_pseudo_children(doc, node, ctx, depth, &[node.after])` — `::after` 疑似

これで paint order が generated/source order に揃い、`::after` overlay が direct abs siblings の上に正しく描画されます。

File: crates/fulgur/src/pageable.rs [hunk 2]
```diff
@@ -1366,20 +1461,106 @@ fn compute_padding_box_inner_radii(outer: &[[f32; 2]; 4], borders: &[f32; 4]) ->
...
+            first.push(clone_pc_with_offset(pc, 0.0));
+            second.push(clone_pc_with_offset(pc, second_y_offset));
+        } else if i < within_idx {
+            first.push(clone_pc_with_offset(pc, 0.0));
+        } else if i > within_idx {
+            second.push(clone_pc_with_offset(pc, second_y_offset));
+        }
+    }
+    (first, second)
+}
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/pageable.rs:1552]: d338672 で修正しました。`split_children_for_within` のシグネチャを変更し、`first_part` / `second_part` を引数として受け取るようにしました。`children` を 1 回 enumerate でイテレートし、`i == within_idx` の位置で split fragment を inject するため、OOF sibling の DOM 順 (within_idx の前後) が両半分で保持されます。call site (split + split_boxed の WithinChild) も同じ呼び出しに統一しています。

unit test `split_children_for_within_inserts_fragments_at_dom_slot` を更新して、`first_part` が within slot 位置 (in-flow と out_of_flow の間) に配置されることをアサートするよう変更しました。

File: crates/fulgur/tests/abs_positioned_pagination.rs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        })
+        .margin(Margin::uniform(0.0))
+        .build();
+    let pdf = engine.render_html(html).expect("render");
+    assert_eq!(page_count(&pdf), 1, "expected one page");
+    // The body content has 'before' and 'after'. If the abs got dropped
+    // by flatten + collect_positioned_children skip, we'd at least still
+    // see the in-flow text. The crucial behavioural check is that the
+    // engine doesn't panic and the document is non-empty.
+    assert!(!pdf.is_empty(), "PDF must not be empty");
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/tests/abs_positioned_pagination.rs]: d338672 で修正しました。test oracle を `engine.build_pageable_for_testing_no_gcpm(html)` で取得した Pageable tree の構造検査に変更し、ヘルパ `count_blocks_with_size_and_flow(tree, 30.0, 30.0, true)` で 30×30 pt の `out_of_flow=true` な `BlockPageable` (= abs `<div>`) が tree に存在することを assert するようにしました。

これで abs child が flatten 経路で drop された場合は count が 0 になり test が fail します (`!pdf.is_empty()` のような krilla の serialisation を介した間接 oracle ではなく、convert path の直接出力を見るため)。

加えて、`display:contents` ラッパは Blitz が abs descendants を完全に layout-tree から落とす別の問題があったため、test fixture を `width:0; height:0; overflow:visible` の通常 `<div>` に変更しました。これが `collect_positioned_children` の flatten 分岐 (ch=0, cw=0, !child_effective_is_empty) に確実に到達する形です。
